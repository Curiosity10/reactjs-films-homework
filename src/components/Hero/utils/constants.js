export const MAX_OVERVIEW_LENGTH = 220;
export const DEFAULT_MOVIE_RUNTIME = '0h 0m';
