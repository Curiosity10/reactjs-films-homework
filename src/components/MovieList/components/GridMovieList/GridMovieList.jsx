import React, { useState } from 'react';
import GridMovieCard from './components/GridMovieCard';

const GridMovieList = ({ movies }) => {
  const [activeCard, setActiveCard] = useState(0);
  return movies.map(movie => (
    <GridMovieCard
      activeCard={activeCard}
      changeActiveCard={id => setActiveCard(id)}
      key={movie.id}
      movie={movie}
    />
  ));
};

export default GridMovieList;
