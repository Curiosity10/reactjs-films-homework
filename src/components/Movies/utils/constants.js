export const MAX_OVERLAY_CARD_DESC_LENGTH = 180;
export const MAX_TABLE_CARD_DESC_LENGTH = 500;
export const MAX_SCROLL_HEIGHT = 0.99;
