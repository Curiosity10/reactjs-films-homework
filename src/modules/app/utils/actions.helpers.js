export const createRequestAction = requestName => ({
  type: requestName,
});

export const createFailureAction = (requestName, error) => ({
  type: requestName,
  payload: {
    error: error.message,
  },
});

export const createReceiveMoviesAction = ({
  type, data, page, genre = null, query = '',
}) => ({
  type,
  payload: {
    movies: data.results,
    currentPage: page,
    pagesCount: data.total_pages,
    genreId: genre,
    searchQuery: query,
  },
});

export const createAsyncAction = ({
  type, asyncFn, page = 1, genre = null, query = '', filterType = '',
}) => (dispatch) => {
  dispatch(createRequestAction(`${type}_REQUEST`));
  return asyncFn({ page, genre, query }, filterType)
    .then(response => response.json())
    .then(data => dispatch(createReceiveMoviesAction({
      type: `${type}_SUCCESS`, data, page, genre, query,
    })))
    .catch(error => dispatch(createFailureAction(`${type}_FAILURE`, error)));
};
