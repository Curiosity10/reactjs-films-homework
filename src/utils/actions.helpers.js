export const createAsyncFnRequest = requestName => ({
  type: requestName,
});

export const createAsyncFnFailure = (requestName, error) => ({
  type: requestName,
  payload: {
    error: error.message,
  },
});

const createReceiveMovies = (type, data, page) => ({
  type,
  payload: {
    movies: data.results,
    currentPage: page,
    pagesCount: data.total_pages,
  },
});

export const createAsyncFnSuccess = (type, asyncFn, page = 1) => (dispatch) => {
  dispatch(createAsyncFnRequest(`${type}_REQUEST`));
  return asyncFn({ page })
    .then(response => response.json())
    .then(data => dispatch(createReceiveMovies(`${type}_SUCCESS`, data, page)))
    .catch(error => dispatch(createAsyncFnFailure(`${type}_FAILURE`, error)));
};
