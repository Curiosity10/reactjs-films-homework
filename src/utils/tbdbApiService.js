import movie from '../assets/json/movie.json';
import genres from '../assets/json/genres.json';
import popular from '../assets/json/popular.json';

const urlBase = 'https://image.tmdb.org/t/p/original';
const urlMinImg = 'https://image.tmdb.org/t/p/w300';

function getNormalRuntime(time) {
  const hours = Math.floor(time / 60);
  const minutes = time - 60 * hours;
  return `${hours}h ${minutes}m`;
}

function getMovieGenres(genresIds) {
  const genresArr = [];
  genresIds.map(genre => genres.genres.map((item) => {
    if (item.id === genre) {
      return genresArr.push({ id: item.id, name: item.name });
    }
    return 0;
  }));
  return genresArr;
}

export function getMovieInfo() {
  return {
    title: movie.original_title,
    genres: movie.genres,
    rating: movie.vote_average,
    runtime: getNormalRuntime(movie.runtime),
    overview: movie.overview,
    image: `${urlBase}${movie.backdrop_path}`,
  };
}

export function getGenres() {
  return genres.genres;
}

export function getPopularMovies() {
  return popular.results.map(item => ({
    id: item.id,
    rating: item.vote_average,
    title: item.title,
    genres: getMovieGenres(item.genre_ids),
    overview: item.overview,
    image: `${urlMinImg}${item.poster_path}`,
  }));
}
