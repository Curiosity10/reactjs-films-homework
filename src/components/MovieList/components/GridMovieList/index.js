import GridMovieList from './GridMovieList';

export default GridMovieList;
