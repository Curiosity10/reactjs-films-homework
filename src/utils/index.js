import {
  getTopMovies,
  searchMovies,
  getGenres,
  getMoviesByGenre,
  getLatestMovies,
  getUpcomingMovies,
  getMovieVideo,
} from './api';
import { getMovies } from './tbdbApiService';
import truncate from './truncate';
import {
  createFailureAction,
  createRequestAction,
  createAsyncAction,
  createReceiveMoviesAction,
} from './actions.helpers';

export {
  getTopMovies,
  searchMovies,
  getGenres,
  getMoviesByGenre,
  getLatestMovies,
  getUpcomingMovies,
  getMovieVideo,
  truncate,
  getMovies,
  createFailureAction,
  createRequestAction,
  createAsyncAction,
  createReceiveMoviesAction,
};
