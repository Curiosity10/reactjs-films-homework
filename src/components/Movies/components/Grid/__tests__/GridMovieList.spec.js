import React from 'react';
import { act, create } from 'react-test-renderer';
import GridMovieList from '..';
import { getMovieInfo } from '../../../../../utils/tbdbApiService';

describe('Grid component renders correctly', () => {
  const movies = [{ ...getMovieInfo(), id: 2 }];

  it('Grid renders correctly', () => {
    const tree = create(<GridMovieList
      movies={movies}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Change active card works correctly', () => {
    const tree = create(<GridMovieList movies={movies} />);
    const mockFunction = jest.fn(tree.toTree().rendered.props.changeActiveCard);
    act(() => {
      mockFunction();
    });
    expect(mockFunction).toHaveBeenCalled();
  });
});
