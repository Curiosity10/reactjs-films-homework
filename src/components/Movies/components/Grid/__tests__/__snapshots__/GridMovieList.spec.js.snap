// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component renders correctly Grid renders correctly 1`] = `
<article
  className="card"
>
  <img
    alt="Poster"
    height="75%"
    src="https://image.tmdb.org/t/p/original/59qyrVGLLtrcuFtJGdixzn7H4OI.jpg"
  />
  <div
    className="onHover"
  >
    <button
      aria-label="Watch"
      className="btnWatch"
      type="button"
    >
      <span>
         Watch Now 
      </span>
    </button>
    <button
      aria-label="View Info"
      className="btn ghost"
      onClick={[Function]}
      type="button"
    >
      View Info
    </button>
  </div>
  <div
    className="info"
  >
    <div>
      <h3
        className="title"
      >
        Ð¢-34
      </h3>
      <p>
        <span
          className="genre"
        >
          War
        </span>
        <span
          className="genre"
        >
          Drama
        </span>
        <span
          className="genre"
        >
          Adventure
        </span>
        <span
          className="genre"
        >
          Action
        </span>
      </p>
    </div>
    <span
      className="numRating"
    >
      4.7
    </span>
  </div>
</article>
`;
